### PR TITLE
Run full Valero conversion for subjects fCamk1 and fCamk2

### PR DESCRIPTION
**Pull request by h-mayorquin**
This is the final debuggin before doing the full conversion. I got my hands on 70 per cent of the data. I should soon have access to all of it (still downloading it). This is a PR for preparing for that conversion it contains the following:

1) Final additions of metadata for the UP and Down states.
2) Warnings instead of assertions when certain files are not found.
3) Modification of `run_conversion` functions to `add_to_nwbfile` in line with the latest changes in neuroconv.
4) First draft of the `convert_all_sessions.py` script.
5) Addition of edge cases in certain interfaces (for example, the SleepIntervals interface sometimes receives only one interval and then requires special logic for that.

Todo:
* [x] Run conversion stubs for all sessions with subjects `fCamk1` and `fCamk1`.
* [x] Debug for errors
* [x] run inspector
* [x] Run full conversion to see if there are memory problems.
* [ ] Get a sense of conversion time and compression ratios.
* Something else?

### CODE REVIEW

**Comment by h-mayorquin** [comment]
The inspector output for the sessions here:

```
Found 62 issues over 12 files:
      14 - BEST_PRACTICE_VIOLATION
      48 - BEST_PRACTICE_SUGGESTION
**************************************************


0  BEST_PRACTICE_VIOLATION
==========================

0.0  fCamk1_200827_sess9.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200827_sess9/fCamk1_200827_110712/Basler_acA1280-60gc__21606137__20200827_110730202.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.1  fCamk1_200827_sess9.nwb and 1 other file: check_time_interval_time_columns - 'TimeIntervals' object with name 'SleepStates'
       Message: ['stop_time'] are time columns but the values are not in ascending order. All times should be in seconds with respect to the session start time.

0.2  fCamk1_200901_sess12.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200901_sess12/fCamk1_200901_115118/Basler_acA1280-60gc__21606137__20200901_115140678.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.3  fCamk1_200902_sess13.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200902_sess13/fCamk1_200902_115511/Basler_acA1280-60gc__21606137__20200902_115537938.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.4  fCamk1_200904_sess15.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200904_sess15/fCamk1_200904_120435/Basler_acA1280-60gc__21606137__20200904_120507437.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.5  fCamk1_200908_sess16.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200908_sess16/fCamk1_200908_124143/Basler_acA1280-60gc__21606137__20200908_124220793.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.6  fCamk1_200909_sess17.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200909_sess17/fCamk1_200909_115825/Basler_acA1280-60gc__21606137__20200909_115853843.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.7  fCamk1_200910_sess18.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200910_sess18/fCamk1_200910_131411/Basler_acA1280-60gc__21606137__20200910_131440006.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.8  fCamk1_200911_sess19.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk1/fCamk1_200911_sess19/fCamk1_200911_112340/Basler_acA1280-60gc__21606137__20200911_112406258.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.9  fCamk2_201012_sess1.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk2/fCamk2_201012_sess1/fCamk2_201012_130347/Basler_acA1280-60gc__21606137__20201012_130405941.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.10  fCamk2_201013_sess2.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk2/fCamk2_201013_sess2/fCamk2_201013_105930/Basler_acA1280-60gc__21606137__20201013_105946653.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.11  fCamk2_201014_sess3.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk2/fCamk2_201014_sess3/fCamk2_201014_112432/Basler_acA1280-60gc__21606137__20201014_112455106.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

0.12  fCamk2_201015_sess4.nwb: check_image_series_external_file_relative - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: The external file '/media/heberto/One Touch/Buzsaki/ValeroM/fCamk2/fCamk2_201015_sess4/fCamk2_201015_110238/Basler_acA1280-60gc__21606137__20201015_110255856.avi' is not a relative path. Please adjust the absolute path to be relative to the location of the NWBFile.

1  BEST_PRACTICE_SUGGESTION
===========================

1.13  fCamk1_200827_sess9.nwb and 11 other files: check_description - 'VectorData' object with name 'gain_to_uV'
        Message: Description ('no description') is a placeholder.

1.14  fCamk1_200827_sess9.nwb and 11 other files: check_description - 'VectorData' object with name 'offset_to_uV'
        Message: Description ('no description') is a placeholder.

1.15  fCamk1_200827_sess9.nwb and 11 other files: check_description - 'VectorData' object with name 'channel_name'
        Message: Description ('no description') is a placeholder.

1.16  fCamk1_200827_sess9.nwb and 11 other files: check_description - 'ProcessingModule' object with name 'behavior'
        Message: Description ('No description.') is a placeholder.
```

File: buzsaki_lab_to_nwb/valero/behaviorinterface.py [hunk 1]
```diff
@@ -170,17 +168,24 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = Fal
...
                 table_rows.append(row_as_dict)
 
+        time_intervals = TimeIntervals(name="SleepStates", description=description)
+        time_intervals.add_column(name="label", description="Sleep state.")
         sorted_table = sorted(table_rows, key=lambda x: (x["start_time"], x["stop_time"]))
-        [table.add_row(**row_as_dict) for row_as_dict in sorted_table]
-        processing_module.add(table)
+        [time_intervals.add_row(**row_as_dict) for row_as_dict in sorted_table]
+
+        processing_module = get_module(nwbfile=nwbfile, name="behavior")
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/valero/behaviorinterface.py:190]:
```suggestion
        processing_module = get_module(nwbfile=nwbfile, name="behavior", description="Contains processed behavioral data.")
```

Or `"Contains behavioral data concerning classified states."` (currently what it is on the Huszar side)

from past references https://github.com/search?q=repo%3Acatalystneuro%2Fbuzsaki-lab-to-nwb%20get_module&type=code

Should probably have a set of defaults like that stored and set in `get_module` depending on the `name`

**Comment by CodyCBakerPhD** [comment]
For the 'not ascending' check, are you using the dev branch of inspector or the latest release?

**Comment by CodyCBakerPhD** [comment]
Just to make sure the DANDI filenames look OK and such, and get rid of the relative path errors, can you try `dandi organize` and then rerun and share the inspection?

**Comment by h-mayorquin** [comment]
I am using dev for inspector, the error is related to this:

https://github.com/NeurodataWithoutBorders/nwbinspector/issues/375

Now I am testing if Neuroscope will be enough if I wll need the new `CellExplorerRecordingInterface` coupled with https://github.com/SpikeInterface/spikeinterface/pull/1741

**Comment by CodyCBakerPhD** [comment]
> I am using dev for inspector, the error is related to this:
 https://github.com/NeurodataWithoutBorders/nwbinspector/issues/375

Ah, right - I remember seeing the issue, but I thought a fix was pushed through? Or no?

File: buzsaki_lab_to_nwb/valero/behaviorinterface.py (hunk 1 above)
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/valero/behaviorinterface.py]: Added something along those lines.

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD 

I ran dandi organize and then inspector before, those are the errors in the post-organize folder:

```
0  BEST_PRACTICE_VIOLATION
==========================

0.0  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 1 other file: check_time_interval_time_columns - 'TimeIntervals' object with name 'SleepStates'
       Message: ['stop_time'] are time columns but the values are not in ascending order. All times should be in seconds with respect to the session start time.

0.1  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=8284.673333333332 and rate=0.03333333333466726 instead of timestamps.

0.2  sub-fCamk1/sub-fCamk1_ses-fCamk1-200901-sess12_behavior+ecephys+image+ogen.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=8093.441333333334 and rate=0.03333333333284827 instead of timestamps.

0.3  sub-fCamk1/sub-fCamk1_ses-fCamk1-200902-sess13_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=11471.841333333334 and rate=0.03333333333466726 instead of timestamps.

0.4  sub-fCamk1/sub-fCamk1_ses-fCamk1-200904-sess15_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=10659.585333333333 and rate=0.03333333333466726 instead of timestamps.

0.5  sub-fCamk1/sub-fCamk1_ses-fCamk1-200908-sess16_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=12492.321333333333 and rate=0.03333333333466726 instead of timestamps.

0.6  sub-fCamk1/sub-fCamk1_ses-fCamk1-200909-sess17_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=10546.049333333332 and rate=0.03333333333466726 instead of timestamps.

0.7  sub-fCamk1/sub-fCamk1_ses-fCamk1-200910-sess18_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=15388.193333333333 and rate=0.03333333333466726 instead of timestamps.

0.8  sub-fCamk1/sub-fCamk1_ses-fCamk1-200911-sess19_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=9474.369333333332 and rate=0.03333333333466726 instead of timestamps.

0.9  sub-fCamk2/sub-fCamk2_ses-fCamk2-201012-sess1_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
       Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=6988.801333333334 and rate=0.03333333333284827 instead of timestamps.

0.10  sub-fCamk2/sub-fCamk2_ses-fCamk2-201013-sess2_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=7750.337333333334 and rate=0.03333333333284827 instead of timestamps.

0.11  sub-fCamk2/sub-fCamk2_ses-fCamk2-201014-sess3_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=7674.337333333334 and rate=0.03333333333284827 instead of timestamps.

0.12  sub-fCamk2/sub-fCamk2_ses-fCamk2-201015-sess4_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=8183.617333333334 and rate=0.03333333333284827 instead of timestamps.

0.13  sub-fCamk3/sub-fCamk3_ses-fCamk3-201111-sess20_behavior+ecephys+image.nwb: check_regular_timestamps - 'ImageSeries' object at location '/acquisition/ImageSeriesTrackingVideo1'
        Message: TimeSeries appears to have a constant sampling rate. Consider specifying starting_time=5968.321333333333 and rate=0.03333333333284827 instead of timestamps.

1  BEST_PRACTICE_SUGGESTION
===========================

1.14  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 27 other files: check_description - 'ElectrodeGroup' object at location '/general/extracellular_ephys/Group 4'
        Message: Description ('no description') is a placeholder.

1.15  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 27 other files: check_description - 'ElectrodeGroup' object at location '/general/extracellular_ephys/Group 3'
        Message: Description ('no description') is a placeholder.

1.16  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 27 other files: check_description - 'ElectrodeGroup' object at location '/general/extracellular_ephys/Group 2'
        Message: Description ('no description') is a placeholder.

1.17  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 27 other files: check_description - 'ElectrodeGroup' object at location '/general/extracellular_ephys/Group 1'
        Message: Description ('no description') is a placeholder.

1.18  sub-fCamk1/sub-fCamk1_ses-fCamk1-200827-sess9_behavior+ecephys+image+ogen.nwb and 27 other files: check_description - 'VectorData' object with name 'gain_to_uV'
        Message: Description ('no description') is a placeholde
```

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**